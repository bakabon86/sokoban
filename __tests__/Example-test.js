import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../src/screens/Login';
import formPushScreen from '../src/screens/formPush';

// test('renders correctly', () => {
//   const tree = renderer.create(<LoginScreen />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// it('app calculateSum returns value', () => {
//     let componentInstance = renderer.create(<App />).getInstance();
//     result = componentInstance.calculatSum(1,3);
//     expect(result["props"]["children"]).toEqual(4);
//   });

// it('should return an array containing the populated options', () => {
//     const teams = [
//       { "name": "A", "id": 1 },
//       { "name": "B", "id": 2 }
//     ];
//     const result = Helper.buildOptions(teams);
  
//     // Check if option exist and have length of 2
//     expect(result).toBeTruthy();
//     expect(result).toHaveLength(2);
  
//     // Check if the options is as expected
//     expect(result[0]).toEqual(<option key="optId1" value={1}>A</option>);
//     expect(result[1]).toEqual(<option key="optId2" value={2}>B</option>);
//   });

test('check user', () => {
    let componentInstance = renderer.create(<LoginScreen />).getInstance();
    result = componentInstance.checkUser();    
    expect(result).toEqual(expect.objectContaining({}));
});
it('Form Push render', () => {
    const tree = renderer.create(<formPushScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
it('works', () => {
    expect(1).toBe(1);
});