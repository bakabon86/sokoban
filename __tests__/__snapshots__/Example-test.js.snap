// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Land Acquisition render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 44,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 30,
          },
          undefined,
        ]
      }
    >
      Land Acquisition
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Tanggal
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "width": 142,
          },
          Object {
            "height": 40,
            "width": 260,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#aaa",
                "borderWidth": 1,
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#fff",
                "marginLeft": 36,
              },
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#c9c9c9",
                },
                undefined,
              ]
            }
          >
            select date
          </Text>
        </View>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 32,
                "marginLeft": 5,
                "marginRight": 5,
                "width": 32,
              },
              Object {
                "left": 0,
                "marginLeft": 0,
                "position": "absolute",
                "top": 4,
              },
            ]
          }
        />
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "portrait-upside-down",
              "landscape",
              "landscape-left",
              "landscape-right",
            ]
          }
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "backgroundColor": "#00000077",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    pointerEvents="auto"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderTopColor": "#ccc",
                            "borderTopWidth": 1,
                            "marginTop": 42,
                          },
                          undefined,
                        ]
                      }
                    >
                      <RCTDatePicker
                        date={1537931632167}
                        locale={undefined}
                        maximumDate={1924880400000}
                        minimumDate={1356973200000}
                        minuteInterval={undefined}
                        mode="date"
                        onChange={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "height": 216,
                          }
                        }
                        timeZoneOffsetInMinutes={null}
                      />
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 42,
                          "justifyContent": "center",
                          "paddingHorizontal": 20,
                          "position": "absolute",
                          "top": 0,
                        },
                        Object {
                          "left": 0,
                        },
                        undefined,
                      ]
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#46cf98",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#666",
                          },
                          undefined,
                        ]
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 42,
                          "justifyContent": "center",
                          "paddingHorizontal": 20,
                          "position": "absolute",
                          "top": 0,
                        },
                        Object {
                          "right": 0,
                        },
                        undefined,
                      ]
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#46cf98",
                            "fontSize": 16,
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Nama
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="User Name"
      style={
        Object {
          "backgroundColor": "#fff",
          "marginBottom": 3,
          "width": 260,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "top": 10,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#41bce6",
              "borderRadius": 5,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 10,
              "width": 260,
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              null,
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
